walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,50 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
 
 public final class FindMeetingQuery {
+
+  public final int duration = 0;
+  public final Boolean notInclusive = false;
+  public final Boolean inclusive = true;
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> attendees = request.getAttendees();
+
+    List<TimeRange> availableEventTimes = new ArrayList<>();
+    events.stream()
+          .sorted((firstRange, secondRange) -> firstRange.getWhen().start() - secondRange.getWhen().start())
+          .forEach(e-> {
+            boolean attendee = !Collections.disjoint(e.getAttendees(), attendees);
+            if (attendee) {
+              availableEventTimes.add(e.getWhen());
+            }
+          });
+
+    availableEventTimes.add(TimeRange.fromStartDuration(TimeRange.END_OF_DAY, duration));
+    List<TimeRange> availableMeetingTimes = new ArrayList<TimeRange>();
+    int meetingStartRequest = TimeRange.START_OF_DAY;
+
+    for (int iterator = 0; iterator < availableEventTimes.size() - 1; iterator++) {
+      int timeRangeStart = availableEventTimes.get(iterator).start();
+      TimeRange availableTimeRange = TimeRange.fromStartEnd(meetingStartRequest, timeRangeStart, notInclusive);
```

Comment:
I appreciate the use of a clear variable name for notInclusive, but the current format is nonstandard. There are two options here:
1. Use parameter comments, which would result in
`TimeRange availableTimeRange = TimeRange.fromStartEnd(meetingStartRequest, timeRangeStart, /* inclusive= */ false);`
2. Switch the constants to all capitals: `NOT_INCLUSIVE`

Reply:
Thank you for mentioning this -- I wasn't entirely sure how to make the parameter more understandable as to what false of true is. I'll change my code so that it follows the second option.